File: Courses/PHP Web Basics - май 2019/16.ExerciseEncapsulationInheritance/01.04.MordorCrueltyPlan/FoodFactory.php
<?php


class FoodFactory
{
    /**
     * @var string
     */
    private $points;

    public function __construct(string $food)
    {
        $this->setPoints($food);
    }

    public function getPoints() : string
    {
        return $this->points;
    }

    public function setPoints($food){
        if ($food == "cram"){
            $this->points = 2;
        }elseif ($food == "lembas"){
            $this->points = 3;
        }elseif ($food == "apple"){
            $this->points = 1;
        }elseif ($food == "melon"){
            $this->points = 1;
        }elseif ($food == "honeycake"){
            $this->points = 5;
        }elseif ($food == "mushrooms"){
            $this->points = -10;
        }else{
            $this->points = -1;
        }
    }
}

File: Courses/C# OOP/Basics/10. Inheritance Exercise/Exercises_Inheritance/05.MordorCruelPlan/Factories/FoodFactory.cs
using _05.MordorCruelPlan.Foods;
using System;
using System.Collections.Generic;
using System.Text;

namespace _05.MordorCruelPlan.Factories
{
    public class FoodFactory
    {
        public Food CreateFood(string type)
        {
            type = type.ToLower();

            switch (type)
            {
                case "apple":
                    return new Apple();
                case "cram":
                    return new Cram();
                case "honeycake":
                    return new HoneyCake();      
                case "lembas":
                    return new Lembas();
                case "melon":
                    return new Melon();
                case "mushrooms":
                    return new Mushrooms();
                default:
                    return new Junk();
            }
        }
    }
}
